creusot/tests/should_succeed/ghost/pcell.rs

```diff
@@ -0,0 +1,158 @@
+extern crate creusot_contracts;
+use creusot_contracts::*;
+
+pub struct Id(()); // TODO: use a why3 abstract type instead ?
+
+impl Id {
+    /// Generate a fresh id.
+    ///
+    /// To ensure that it is different from previously generated ids, see [`Self::differentiate`].
+    #[pure]
+    #[requires(true)]
+    #[ensures(true)]
+    fn fresh() -> Self {
+        Self(())
+    }
```

Comment:
What is our position on non-determinism in ghost code?

Reply:
I would indeed have thought that ghost code should be deterministic. At some point we have imagined that ghost functions could be turned into logical functions under appropriate assumptions.